python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -137,6 +137,10 @@ def from_assets(
     def all_asset_keys(self) -> AbstractSet[AssetKey]:
         return self._asset_dep_graph["upstream"].keys()
 
+    @property
+    def non_source_asset_keys(self) -> AbstractSet[AssetKey]:
```

Comment:
we have such an awful ugly trinary thing going on (SourceAsset, AssetsDef, thing that is referenced by an AssetsDef but has no SourceAsset) but this works